src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,550 @@
+        raise ValueError(f"{assembly_dir} is not a symlink")
+    src_file = os.path.join(os.readlink(assembly_dir), assembly_file)
+    return src_file
+
+
+def _upload_assembly_to_workspace(
+    conf: Conf,
+    workspace_name: str,
+    file_path: str,
+    assembly_name: str,
+) -> None:
+    """Upload an assembly file to workspace."""
+    success, attempts, max_attempts = False, 0, 3
+    while attempts < max_attempts and not success:
+        try:
+            time.sleep(attempts)
+            conf.asu.save_assembly_from_fasta(
+                {
+                    "file": {"path": file_path},
+                    "workspace_name": workspace_name,
+                    "assembly_name": assembly_name,
+                }
+            )
+            success = True
+        except Exception as e:
+            print(f"Error:\n{e}\nfrom attempt {attempts + 1}.\nTrying to rerun.")
+            attempts += 1
+
+    if not success:
+        raise ValueError(
+            f"Upload Failed for {file_path} after {max_attempts} attempts!"
+        )
+
+
+def _read_yaml_file(
+        root_dir: str,
+        workspace_id: int,
+        env: str
+) -> Tuple[dict[str, dict[int, list[str]]], list[str]]:
+    """
+    Get all assembly names from the uploaded.yaml file.
+    """
+
+    if env not in loader_common_names.KB_ENV:
+        raise ValueError(f"Currently only support these {loader_common_names.KB_ENV} envs for upload")
+
+    file_path = _get_yaml_file_path(root_dir)
+
+    with open(file_path, "r") as file:
+        data = yaml.safe_load(file)
+
+    if not data:
+        data = {env: dict()}
+
+    if workspace_id not in data[env]:
+        data[env][workspace_id] = list()
+
+    assembly_names = data[env][workspace_id]
+    return data, assembly_names
+
+
+def _update_yaml_file(
+        root_dir: str,
+        workspace_id: int,
+        env: str,
+        new_assemly_names: list[str],
+        overwrite: bool = False,
+) -> list[str]:
+    """
+    Update the uploaded.yaml file with newly uploaded assemblies.
+    """
+    data, assembly_names = _read_yaml_file(root_dir, workspace_id, env)
+
+    if not overwrite:
+        intersection = set(assembly_names).intersection(set(new_assemly_names))
+        if intersection:
+            raise ValueError(f"Detected assembly names {intersection} waiting to be uploaded "
+                             f"already exist in the uploaded.yaml file")
+        for new_assembly_name in new_assemly_names:
+            data[env][workspace_id].append(new_assembly_name)
+    else:
+        all_assembly_names = list(set(assembly_names).union(set(new_assemly_names)))
+        data[env][workspace_id] = all_assembly_names
+
+    file_path = _get_yaml_file_path(root_dir)
+    with open(file_path, "w") as file:
+        yaml.dump(data, file)
+
+    assembly_names = data[env][workspace_id]
+    return assembly_names
+
+
+def _fetch_assemblies_to_upload(
+    workspace_name: str,
+    csd: str,
```

Comment:
Could you give a more descriptive name than `csd`? There's a number of places this name is used that could be updated

Reply:
Updated globally!

Reply:
:+1: